### ISSUE TITLE
SessionId missing in response

### ISSUE THREAD

**Issue by abteilung6** [Sentiments-of-Bundestag/graphenauswertung#23]
Regarding your endpoint we miss the addtional field sessionId.

 /factions/ranked
```
{
      "recipient": "F001",
      "sender": "F000",
      "sentiment": -1.399999976158142,
      "count": 10
      // missing
      sessionId: ""
    },
```
/factions/ranked
```
  {
    "name": "CDU/CSU", 
    "size": 246,
    "factionId": "F000",
    "rank": "0.7"
      // missing
      sessionId: ""
  }, 

etc.
```

Our interface expectations:
```typescript

// Sentiment

// Completly Missing
export interface Session {
  sessionId: number;
  legislativePeriod: number;
  endDateTime: string;
  startDateTime: string;
}

export interface Faction {
  // Missing
  factionId: string;
  // Missing
  sessionId: number;
  name: string;
  size: number;
}

export interface FactionGraph {
  recipient: string;
  // Missing
  sessionId: number;
  sender: string;
  sentiment: number;
  // Missing
  count: number;
}

export interface FactionRanked {
  // Missing
  factionId: string;
  // Missing
  sessionId: string;
  name: string;
  size: number;
  rank: number;
}

export interface Person {
  speakerId: string;
  // Missing
  sessionId: number;
  name: string;
  rank: number;
  role: string;
}

// Completly missing
export interface PersonMessage {
  // Missing
  sessionId: number;
  sender: string;
  recipient: string;
  sentiment: number;
}

// Completly missing
export interface PersonGraph {
  // missing
  sessionId: number;
  sender: string;
  recipient: string;
  sentiment: number;
  count: number;
}

export interface PersonRanked {
  speakerId: number;
  // missing
  factionId: string;
  faction: string;
  name: string;
  role: string;
  rank: number;
}


```

**Comment by glutengo** [comment]
Thanks for your input. I understand most of the requirements and most of them are already in porgress.

Questions:
* Why do the nodes (Person / Faction) need a sessionId? These entities are not directly related to the parliament sessions, only the messages are.
* What is the difference between PersonGraph and PersonMessage? Is it only the additional count property? If so, I would like to keep things simple and only provide a single type which includes the count.

We noticed this morning that the group 4 database does currently not have any messages anymore, so I suppose, they are currently working on something there.

**Comment by glutengo** [comment]
@abteilung6 

As mentioned in the other issues, we have deployed an updated version today. The endpoints are in line with the readme on the develop branch (https://github.com/Sentiments-of-Bundestag/graphenauswertung#endpoints).

I have also spoken to Group 4 and Group 5 about a few open issues:
* Group 4 have started to import more data. This becomes visible when accessing our new endpoint http://infosys6.f4.htw-berlin.de/sessions because this returns more than one entry now. As you will notice, some of our other endpoints will take longer now because the amount of data is way bigger than it was before (~ 30k comments vs. ~300 comments). We will add some caching functionality to improve this in the future as we laid out in the planning presentation.
* Group 4 have not finished importing all the data yet. They expect ~200k comments and are currently struggling with processing all the data in a single import job. It is therefore expected that there may be the odd moment when the database is empty or contains less data in the next days because they are still working on it
* Group 5 have not provided us with an option to filter the data by a sessionId or even a date. We are still waiting for their feedback on this content. We have already built the option to filter the faction-based graph by a session id but this results in graphs without any messages at the moment because we do not have the correct dates on the messages in the group 5 database yet

I hope this helps you in some way. Please keep asking if there is anything else I can help with.

**Comment by glutengo** [comment]
As it seems, Group 4 seem still to be working on the import. They now have about 70k comments in there but they have completely changed their model (different node names, relationship names, attribute names etc.) so our application is not compatible with that anymore. I will speak to them and give an update here once this is resolved.

**Comment by glutengo** [comment]
will be resolved in #26
